src/Model/MediaInterface.php

```diff
@@ -380,4 +381,8 @@ public function getGalleryItems();
      * @return string
      */
     public function getPreviousProviderReference();
+
+    public function getCategory(): ?CategoryInterface;
```

Comment:
Should we have another Interface `MediaWithCategoryInterface` with these two methods instead ?

If you don't use the ClassificationnBundle you need MediaInterface, if you use the ClassificationBundle, you'll need the new Interface.

Reply:
I was thinking the same, but you will normally extend the src/Entity/BaseMedia.php, so you will always get the one with get/set Category.

We can add that interface, but the src/Entity will implement both.

Reply:
I would have said that if someone wants the media without the classification, he could have implement the interface so 2 interfaces seems better than 1.

But...
Whoah, this code https://github.com/sonata-project/SonataMediaBundle/blob/master/src/Model/Media.php#L17 mean that the Media entity cannot be used without the ClassificationBundle, isn't it ?
I don't understand how the "split" could work if the class is used.

If we use the Classification classes in the "root code" of the MediaBundle, I don't see other solution than adding the ClassificationBundle in the require section (and not only the require-dev). Or I totally misunderstand how this bundle works.

Reply:
This bundle works without the ClassificationBundle (I am using it on my projects).

The thing is: If you don't install classification bundle, the relation with media and Category is not created: https://github.com/sonata-project/SonataMediaBundle/blob/3.x/src/DependencyInjection/SonataMediaExtension.php#L627-L638

Then, you just don't use the setCategory and getCategory, and that's all. The fields are not displayed on the admin pages.